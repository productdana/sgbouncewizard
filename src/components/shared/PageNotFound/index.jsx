import React from "react";
import { Link } from "react-router-dom";
import { Row } from "../Row";
import { Column } from "../Column";
import "./index.scss";

const PageNotFound = () => (
  <div className="not-found-container">
    <Row className="logo-container sg-center">
      <Column width={2} offset={6}>
        <img
          data-test="err-logo"
          className="err-logo sg-center"
          src="https://sendgrid.com/brand/sg-logo-white.png"
          alt="sg-logo"
        />
      </Column>
      <Row data-test="err-msg">
        <Column width={4} offset={5}>
          <h1>404: Page Not Found</h1>
        </Column>
        <div className="space-1" />
        <Column width={4} offset={5}>
          <p className="not-found-subtext sg-center">
            {"Looks like you've fallen off the grid."}
          </p>
        </Column>
        <Column width={4} offset={5}>
          <span
            data-test="err-btn"
            className="btn btn-secondary btn-on-dark sg-centers"
          >
            <Link to="/bounce_rules"> {"Go To Home Page"}</Link>
          </span>
        </Column>
      </Row>
    </Row>
  </div>
);

export default PageNotFound;
