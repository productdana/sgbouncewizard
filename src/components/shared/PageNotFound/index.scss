.not-found-container {
  height: 100%;
  background: linear-gradient(
      rgba(26, 130, 226, 0.85),
      rgba(26, 130, 226, 0.85)
    ),
    url(/public/assets/penguin.gif) no-repeat 50% fixed;
  background-size: cover;
  h1 {
    color: #fff;
    text-shadow: 0 2px 3px rgba(41, 70, 97, 0.15);
    margin: 0;
    margin-top: 1rem;
    font-size: 3rem;
  }
  .link,
  a {
    color: #fff;
  }
}

.err-logo {
  width: 9.492rem;
  margin-bottom: 5vh;
}

.logo-container {
  padding-top: 30vh;
  padding-bottom: 3vh;
}

.not-found-subtext {
  font-size: 1.2rem;
  font-weight: 900;
  color: #fff;
  text-shadow: 0 2px 3px rgba(41, 70, 97, 0.15);
  margin-bottom: 8vh;
}

.space-1 {
  height: 1.5rem;
}
