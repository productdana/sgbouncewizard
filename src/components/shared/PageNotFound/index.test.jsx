import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import PageNotFound from ".";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<PageNotFound />);
});

it("should render correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <PageNotFound />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render SG logo", () => {
  expect(wrapper.find("[data-test='err-logo']")).toHaveLength(1);
});

it("should render 404 message", () => {
  expect(wrapper.find("[data-test='err-msg']")).toHaveLength(1);
});

it("should render home button", () => {
  expect(wrapper.find("[data-test='err-btn']")).toHaveLength(1);
});
