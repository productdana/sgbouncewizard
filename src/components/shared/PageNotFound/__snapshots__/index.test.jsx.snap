// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="not-found-container"
>
  <div
    className="mako-grid__row logo-container sg-center"
  >
    <div
      className="mako-grid__col-2 col-start-6"
    >
      <img
        alt="sg-logo"
        className="err-logo sg-center"
        data-test="err-logo"
        src="https://sendgrid.com/brand/sg-logo-white.png"
      />
    </div>
    <div
      className="mako-grid__row"
      data-test="err-msg"
    >
      <div
        className="mako-grid__col-4 col-start-5"
      >
        <h1>
          404: Page Not Found
        </h1>
      </div>
      <div
        className="space-1"
      />
      <div
        className="mako-grid__col-4 col-start-5"
      >
        <p
          className="not-found-subtext sg-center"
        >
          Looks like you've fallen off the grid.
        </p>
      </div>
      <div
        className="mako-grid__col-4 col-start-5"
      >
        <span
          className="btn btn-secondary btn-on-dark sg-centers"
          data-test="err-btn"
        >
          <a
            href="/bounce_rules"
            onClick={[Function]}
          >
             
            Go To Home Page
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
